third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -1009,17 +1009,19 @@ class ConvertSCFIfOp : public PointerCanonicalizationPattern<scf::IfOp> {
       rewriter.inlineBlockBefore(ifOp.elseBlock(), newIfOp.elseBlock(),
                                  newIfOp.elseBlock()->begin());
 
-    rewriter.replaceOpWithMultiple(ifOp, {newIfOp.getResults()});
-
+    auto results = newIfOp.getResults();
+    SmallVector<ValueRange> replacements;
     for (int64_t idx :
          llvm::cast<DenseI64ArrayAttr>(newIfOp.thenYield()->getDiscardableAttr(
                                            kSCFIfOpYieldFatPtrOffsets))
              .asArrayRef()) {
       Value thenFatPtrBase = newIfOp.thenYield().getOperand(idx);
       Value thenFatPtrOffset = newIfOp.thenYield().getOperand(idx + 1);
-      fatPtrs[{newIfOp.getResult(idx), newIfOp.getResult(idx + 1)}] =
+      replacements.push_back({results[idx], results[idx + 1]});
+      fatPtrs[{results[idx], results[idx + 1]}] =
           fatPtrs.at({thenFatPtrBase, thenFatPtrOffset});
```

Comment:
for my own understanding, could you explain why only the thenYield values are considered and not the else ones? Is it because only the type matters?

Reply:
Yup that's exactly correct - the whole analysis is just (roughly) 1:N type conversion so only the types matter (the `fatPtrs` dictionary just tracks attributes which should likewise be the same along both `then` and `else`).

Reply:
Can you add some comments to the code to explain this part given it's not that straightforward.